### DISCUSSION TITLE
Associate DagRun to Kubernetes Pod

### DISCUSSION THREAD

**Discussion by teaglebuilt** [apache/airflow#22316] [Ideas]
**Description**

Run an entire dag on the kubernetes pod operator. There is a label called `reattach on restart` that designates the task back to the pod for execution in the event of a schedule failure, but it would be nice to have this feature in general for all tasks.

**Use case / motivation**

We might want to launch a pod to run an isolated environment for a dagrun but not want a pod spinned up for every task. Its possible that a pipeline might be executing remote processes and not in need of heavy computing resources. So if a pod could be launched as a dagrun, this would help provide multi tenant solutions.

Almost like a sub dag operator,  but with using the Kubernetes Pod Operator. Kedro's Airflow implementation is closer to this concept by using it to launch a pod per node in the pipeline, but that is still representing it a task / step level.

**Are you willing to submit a PR?**

I am willing to contribute to this effort, if deemed so.

**Related Issues**

No, although i am wondering if this would cause bottle necks with the scheduler, but it can scale now...

**Comment by teaglebuilt** [comment]
@eladkal well, if this is a no, can i get some opinion over the value of this feature? If so, I would be willing to take this PR. Although, before hand I would like to see what the community thinks about this...potential issues before hand, opinions..

**Comment by eladkal** [comment]
@dimberman @jedcunningham  WDYT?

**Comment by jedcunningham** [comment]
I'm not opposed to the idea, but I don't think it will be easy to pull off with the way Airflow works currently. I'll defer to @ash and/or @kaxil.
